applications/minotari_console_wallet/src/ui/components/register_template_tab.rs

```diff
@@ -638,29 +638,36 @@ impl<B: Backend> Component<B> for RegisterTemplateTab {
 
         let rx_option = self.result_watch.take();
         if let Some(rx) = rx_option {
-            trace!(target: LOG_TARGET, "{:?}", (*rx.borrow()).clone());
-            let status = match (*rx.borrow()).clone() {
-                UiTransactionSendStatus::Initiated => "Initiated",
-                UiTransactionSendStatus::Error(e) => {
-                    self.error_message = Some(format!("Error sending transaction: {}, Press Enter to continue.", e));
-                    return;
-                },
-                UiTransactionSendStatus::TransactionComplete => {
-                    self.fee_per_gram = app_state.get_default_fee_per_gram().as_u64().to_string();
-                    self.template_name = "".to_string();
-                    self.template_type = "".to_string();
-                    self.binary_url = "".to_string();
-                    self.binary_checksum = "".to_string();
-                    self.repository_url = "".to_string();
-                    self.repository_commit_hash = "".to_string();
-                    self.success_message =
-                        Some("Transaction completed successfully!\nPlease press Enter to continue".to_string());
-                    return;
-                },
-                status => {
-                    warn!("unhandled transaction status {:?}", status);
-                    return;
-                },
+            {
+                // Ensure the watch borrow is dropped immediately after use
+                trace!(target: LOG_TARGET, "{:?}", (*rx.borrow()).clone());
+            }
+            let status = {
+                // Ensure the watch borrow is dropped immediately after use
+                match (*rx.borrow()).clone() {
```

Comment:
The borrow should/may only be alive until the end of  `Clone::clone` (I think rust could keep the borrow alive to prevent any references to the match data from going out of scope too early - it would be interesting to look into that) but even so, all references would go out of scope by the end of the match, so another outer scope is unnecessary.

Simple example here
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ad27efed3397c18f31312783969455d7